### PR TITLE
Move responsibility of handling output files to OutputManager

### PR DESCRIPTION
**Pull request by bartgol**
Moving file management to the OutputManager, we allow writing more than one output stream on the same file (e.g., for both dyn and phys grids output). Before, two separate instances of AtmosphereOutput were both acting as if they owned the file (hence, updating time, closing files,...), which is a problem when 2+ instances write on the same file. Additionally, rpointer files (particularly during unit tests) could easily get corrupted.

Other miscellanea changes:
- bug fixes in the `share/io` folder (one related to padding, one related to the F90 interface)
- added ability to subtract time stamps, to get elapsed seconds.
- storing/loading initial date/tim in nc files. Allows to restart time dimension from where it stopped (rather than start from 0 again).

### CODE REVIEW

File: components/scream/src/share/util/scream_time_stamp.cpp [hunk 1]
```diff
@@ -200,6 +205,72 @@ TimeStamp operator+ (const TimeStamp& ts, const int dt) {
...
+      diff += spd*days_in_month(y1,m);
+    }
+    diff += spd*(d1-1);
+    diff += ts1.sec_of_day();
+  } else if (m1>m2) {
+    // Rest of the day and month in ts2's current month
+    diff += spd - ts2.sec_of_day();
+    diff += spd*(days_in_month(y2,m2)-d2);
+
+    // Whole montsh in (m2,m1)
```
Comment (AaronDonahue) [inline components/scream/src/share/util/scream_time_stamp.cpp:251]: "months"

File: components/scream/src/share/io/scorpio_input.cpp [hunk 2]
```diff
@@ -25,7 +25,11 @@ AtmosphereInput (const ekat::Comm& comm,
                  const std::shared_ptr<const gm_type>& grids_mgr)
  : AtmosphereInput(comm,params)
 {
-  init (field_mgr,grids_mgr);
+  // Sets the intermal field mgr, and possibly sets up the remapper
```
Comment (AaronDonahue) [inline components/scream/src/share/io/scorpio_input.cpp:28]: "internal"

File: components/scream/src/share/io/scorpio_output.hpp [hunk 3]
```diff
@@ -27,7 +27,12 @@
  *  Casename:                     STRING
  *  Averaging Type:               STRING
  *  Max Snapshots Per File:       INT                   (default: 1)
- *  Fields:                       ARRAY OF STRINGS
+ *  Grids:                        ARRAY OF STRINGS
```
Comment (AaronDonahue) [inline components/scream/src/share/io/scorpio_output.hpp:30]: I'm a bit confused here, from the code above I understood that one can also just list "Fields" at the level of "Grids" like is currently done.  Maybe I misunderstood the checks though.

File: components/scream/tests/coupled/homme_physics/homme_shoc_cld_p3_rrtmgp.cpp [hunk 4]
```diff
@@ -63,11 +63,16 @@ TEST_CASE("scream_homme_physics", "scream_homme_physics") {
   //       during RRTMGPRatiation::finalize_impl, after RRTMGP has deallocated
   //       all its arrays.
   ad.initialize(atm_comm,ad_params,t0);
-  printf("Start time stepping loop...       [  0%%]\n");
+
+  if (atm_comm.am_i_root()) {
```
Comment (AaronDonahue) [inline components/scream/tests/coupled/homme_physics/homme_shoc_cld_p3_rrtmgp.cpp:67]: Thank you for this change.

**Review by AaronDonahue — COMMENTED** [review]
Just a couple of questions, but other than that it looks great!  There was a lot of work that went into this and I think it cleans the code up considerably.  Thanks!

File: components/scream/src/share/io/scorpio_output.hpp (hunk 3 above)
Comment (bartgol) [reply components/scream/src/share/io/scorpio_output.hpp:30]: I will go over this documentation, and make sure it's consistent.

Comment (bartgol) [reply components/scream/src/share/io/scorpio_output.hpp:30]: Ah, yeah, it is possible, but the AD (or, better, the OutputManager) was not really supporting it, making it effectively pointless. I'll add this possibility back.

**Review by AaronDonahue — APPROVED** [review]

**Review by AaronDonahue — APPROVED** [review]

**Pull request merged**